### PR TITLE
refactor: extract secure boot certificate generation

### LINKED ISSUE #7412: move gen-certs to talosctl
we can move gen-certs to `talosctl` to make it easier to be used, we can add some flags to customize common name, etc.

### PR DESCRIPTION
**Pull request by smira**
Fixes #7412

### CODE REVIEW

File: cmd/talosctl/cmd/mgmt/gen/secureboot.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+			return err
+		}
+	}
+
+	fmt.Fprintf(os.Stderr, "writing %s\n", path)
+
+	return os.WriteFile(path, data, perm)
+}
+
+func generateSigningCerts(path, prefix, commonName string) error {
```
Comment (frezbo) [inline cmd/talosctl/cmd/mgmt/gen/secureboot.go:102]: we need to also pass in the key bits, pcr certs needs a 2048 key (tpm doesn't allow loading an rsa key with size > 2048)

File: cmd/talosctl/cmd/mgmt/gen/secureboot.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+	return checkedWrite(filepath.Join(path, prefix+"-signing-public-key.pem"), privKey.PublicKeyPEM, 0o600)
+}
+
+// generateSecureBootDabase generates a UEFI database to enroll the signing certificate.
+//
+// ref: https://blog.hansenpartnership.com/the-meaning-of-all-the-uefi-keys/
+//
+//nolint:gocyclo
+func generateSecureBootDabase(path, enrolledCertificatePath, signingKeyPath, signingCertificatePath string) error {
```
Comment (frezbo) [inline cmd/talosctl/cmd/mgmt/gen/secureboot.go:143]:
```suggestion
func generateSecureBootDatabase(path, enrolledCertificatePath, signingKeyPath, signingCertificatePath string) error {
```

File: website/content/v1.5/reference/cli.md [hunk 3]
```diff
@@ -97,7 +97,7 @@ talosctl cluster create [flags]
       --bad-rtc                                  launch VM with bad RTC state (QEMU only)
       --cidr string                              CIDR of the cluster network (IPv4, ULA network for IPv6 is derived in automated way) (default "10.5.0.0/24")
       --cni-bin-path strings                     search path for CNI binaries (VM only) (default [/home/user/.talos/cni/bin])
-      --cni-bundle-url string                    URL to download CNI bundle from (VM only) (default "https://github.com/siderolabs/talos/releases/download/v1.5.0-alpha.1/talosctl-cni-bundle-${ARCH}.tar.gz")
+      --cni-bundle-url string                    URL to download CNI bundle from (VM only) (default "https://github.com/smira/talos/releases/download/v1.5.0-alpha.1/talosctl-cni-bundle-${ARCH}.tar.gz")
```
Comment (frezbo) [inline website/content/v1.5/reference/cli.md:100]: :sweat_smile:

**Review by frezbo — APPROVED** [review]

File: website/content/v1.5/reference/cli.md (hunk 3 above)
Comment (smira) [reply website/content/v1.5/reference/cli.md:100]: :hankey: different machine

File: cmd/talosctl/cmd/mgmt/gen/secureboot.go (hunk 1 above)
Comment (smira) [reply cmd/talosctl/cmd/mgmt/gen/secureboot.go]: fixed, thanks!

File: cmd/talosctl/cmd/mgmt/gen/secureboot.go (hunk 2 above)
Comment (smira) [reply cmd/talosctl/cmd/mgmt/gen/secureboot.go:143]: fixed

File: website/content/v1.5/reference/cli.md (hunk 3 above)
Comment (smira) [reply website/content/v1.5/reference/cli.md]: fixed

**Review by frezbo — APPROVED** [review]

**Pull request merged**